src/server/api/routes/users.js

```diff
@@ -368,4 +410,43 @@ router.delete("/:userId/by-obj-id", (req, res, next) => {
     });
 });
 
+router.post('/:userId/image', (req, res, next) => {
+    if (Object.keys(req.files).length == 0) {
+        return res.status(400).send('No files were uploaded.');
+    }
+
+    let image   = req.files.profileImage;
+    let imgType = /\.[\w\d]+$/.exec(image.name)[0];
+    let imgName = `${req.params.userId}${imgType}`;
+
+    let params = {
+        Body:           image.data,
+        Bucket:         `${process.env.S3_BUCKET_NAME}/${process.env.S3_IMAGE_PATH}`,
+        Key:            imgName,
+        ContentType:    image.mimetype
+    };
+
+    s3.putObject(params, (err, data) => {
+        console.log(data);
+        if (err) {
+            log.error('ERROR HERE', err);
+            res.status(500).json(err);
+            return;
+        }
+        else {
+            const pictureUrl = `https://s3-${process.env.AWS_REGION}.amazonaws.com/${process.env.S3_BUCKET_NAME}/${process.env.S3_IMAGE_PATH}/${imgName}?cacheStop=${Date.now()}`;
+
+            User.update({ userId: req.params.userId },
+                {$set: { picture: pictureUrl }}).exec()
+            .then(() => {
+                res.status(200).json({ picture: pictureUrl });
+            })
+            .catch((err) => {
+                log.error('ERROR HERE 2', err);
```

Comment:
Could use a more descriptive log here :)

Reply:
Bahaha! My bad. I must have been pretty burnt out at this point.

Reply:
Haha all good.